app/src/main/java/org/piramalswasthya/cho/ui/commons/pharmacist/PharmacistFormViewModel.kt

```
@@ -399,4 +413,90 @@ class PharmacistFormViewModel @Inject constructor(
+            emptyList()
+        }
+    }
+
+    private fun calculateExpiryInDays(expiryDate: String): Int {
+        return try {
+            val sdf = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'", Locale.US)
+            val expiry = sdf.parse(expiryDate)
+            val today = Date()
+            val diffInMillis = expiry?.time?.minus(today.time) ?: 0
+            (diffInMillis / (1000 * 60 * 60 * 24)).toInt()
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)